assemblies/assembly-planner-configuration.adoc

```diff
@@ -0,0 +1,1986 @@
+These `Config` classes are the Java representation of the XML format.
+They build the runtime components of the package namespace ``org.optaplanner.core.impl`` and assemble them into an efficient ``Solver``.
+
+[NOTE]
+====
+To configure a `SolverFactory` dynamically for each user request, build a template `SolverConfig` during initialization and copy it with the copy constructor for each user request:
+
+[source,java,options="nowrap"]
+----
+    private SolverConfig template;
+
+    public void init() {
+        template = SolverConfig.createFromXmlResource(
+                "org/optaplanner/examples/nqueens/solver/nqueensSolverConfig.xml");
+        template.setTerminationConfig(new TerminationConfig());
+    }
+
+    // Called concurrently from different threads
+    public void userRequest(..., long userInput) {
+        SolverConfig solverConfig = new SolverConfig(template); // Copy it
+        solverConfig.getTerminationConfig().setMinutesSpentLimit(userInput);
+        SolverFactory<NQueens> solverFactory = SolverFactory.create(solverConfig);
+        Solver<NQueens> solver = solverFactory.buildSolver();
+        ...
+    }
+----
+====
+
+[[annotationAlternatives]]
+
+=== {PLANNER_SHORT} annotation
+
+OptaPlanner needs to know which classes in your domain model are planning entities, which properties are planning variables, and so on. Use one of the following methods to add annotations to your {PLANNER_SHORT} project:
+
+* Add class annotations and JavaBean property annotations on the domain model.  The property annotations must be on the getter method, not on the setter method. Annotated  getter methods do not need to be public.
++
+NOTE: This is the recommended method.
+* Add class annotations and field annotations on the domain model. Annotated fields do not need to be public.
+
+
+[[domainAccess]]
+=== Specifying {PLANNER_SHORT} domain access
+
+By default, OptaPlanner accesses your domain using reflection. Reflection is reliable but slow compared to direct access. Alternatively, you can configure OptaPlanner to access your domain using Gizmo, which will generate bytecode that directly accesses the
+fields and methods of your domain without reflection. However, this method has the following restrictions:
+
+* The planning annotations can only be on public fields and
+  public getters.
+* `io.quarkus.gizmo:gizmo` must be on the classpath.
+
+NOTE: These restrictions do not apply when you use OptaPlanner with Quarkus,
+where Gizmo is the default domain access type.
+
+.Procedure
+To use Gizmo outside of Quarkus, set the `domainAccessType` in the
+Solver Configuration:
++
+[source,xml,options="nowrap"]
+----
+  <solver>
+    <domainAccessType>GIZMO</domainAccessType>
+  </solver>
+----
+
+[[customPropertiesConfiguration]]
+=== Configuring custom properties
+
+Solver configuration elements that instantiate classes and explicitly mention it, support custom properties.
+COMMENT: What does 'explicitly mention it' mean?
```

Comment:
Here it means that the parameter MyEasyScoreCalculator myCacheSize is defined in config though can have several configuration without recompiling the module. The user is free to not define it in the config file, that will make myCacheSize implicit

Reply:
So I propose to mention why it is useful (the comment bellow )

Reply:
I reinstated the community text to the note. Do you want me to add something else as well?

Reply:
No that is ok, thanks